rocketpy/Flight.py

```diff
@@ -1228,8 +1228,19 @@ def uDot(self, t, u, postProcessing=False):
             # Mass
             MtDot = self.rocket.motor.massDot.getValueOpt(t)
             Mt = self.rocket.motor.mass.getValueOpt(t)
+            # Pressures
+            exitPressure = self.rocket.motor.exitPressure
+            freestreamPressure = self.env.pressure(z)
+            # Check for optimal expansion
+            calculatePT = not self.rocket.motor.altitudeCompensatingNozzle
             # Thrust
-            Thrust = self.rocket.motor.thrust.getValueOpt(t)
+            pressureThrust = (
+                calculatePT
+                * (exitPressure - freestreamPressure)
+                * np.pi
+                * self.rocket.motor.nozzleRadius ** 2
+            )
+            Thrust = self.rocket.motor.thrust.getValueOpt(t) + pressureThrust
```

Comment:
In terms of design, is this implementation better than having `motor.thrust` include the pressure term? If so, how?

(In such implementation, the thrust curve supplied by the `Motor` to the `Flight` class would need to be a function of both time and pressure. The code at the `Flight` class would not change significantly.)

Reply:
That's an interesting point which I've thought about quite a bit during the implementation.

My reasoning was that the momentum thrust (currently represented by `motor.thrust`) is an inherent function of the motor itself, while the pressure thrust part will also be dependent on the external environment and on a given trajectory - so it is a bit easier to keep them separate. 

We know exactly how the momentum thrust will behave over time due to thrust curve being an input, while the best we can do for pressure thrust **while still inside motor class** is to the create it as function-defined Function, with no known boundaries. We could create a new `totalThrust` attribute which is a sum of both and a function of two parameters, yet it would inherit the issues that pressureThrust has inside of `Motor`, namely the lack of information about pressure(t). Perhaps it would be more interesting to create it as an attribute of `Flight`.

Another point is that momentum thrust (currently `motor.thrust`) is used on a few other calculations (such as `massDot`) - so it probably should be its own thing (this doesn't prevent a new totalThrust attribute from existing of course).

Reply:
Thanks for the clarifications @MrGribel. It makes a lot of sense.

I agree with you, perhaps a `Flight.totalThrust` and a `Flight.pressureThrust` attribute (as a Function of time) would be usefull.